src/CPUSolver.cpp

```diff
@@ -919,30 +915,7 @@ void CPUSolver::transferAllInterfaceFluxes() {
     }
 
     /* Block for communication round to complete */
-    bool round_complete = false;
-    while (!round_complete) {
-
-      round_complete = true;
-      int flag;
-
-      /* Check forward and backward send/receive messages */
-      for (int i=0; i < num_domains; i++) {
-
-        /* Wait for send to complete */
-        if (_MPI_sends[i] == true) {
-          MPI_Test(&_MPI_requests[i*2], &flag, &stat);
-          if (flag == 0)
-            round_complete = false;
-        }
-
-        /* Wait for receive to complete */
-        if (_MPI_receives[i] == true) {
-          MPI_Test(&_MPI_requests[i*2+1], &flag, &stat);
-          if (flag == 0)
-            round_complete = false;
-        }
-      }
-    }
+    MPI_Waitall(2 * num_domains, _MPI_requests, MPI_STATUSES_IGNORE);
```

Comment:
Great catch ! We currently dont need to initialize since the old value is good if we don't communicate.
But I'm working on a different way to store boundary fluxes, and it becomes necessary to initialize to MPI_REQUEST_NULL